churaverse-plugins-client/src/churarenPlugin/churarenAlchemyItems/trapPlugin/trapPlugin.ts

```diff
@@ -0,0 +1,155 @@
+import { BaseAlchemyItemPlugin } from '@churaverse/churaren-alchemy-plugin-client/domain/baseAlchemyItemPlugin'
+import { TrapPluginStore } from './store/defTrapPluginStore'
+import { initTrapPluginStore, resetTrapPluginStore } from './store/initTrapPluginStore'
+import { Trap, TRAP_ITEM } from './domain/trap'
+import { TrapSpawnMessage } from './message/trapSpawnMessage'
+import { SocketController } from './controller/socketController'
+import { TrapAttackRendererFactory } from './renderer/trapAttackRendererFactory'
+import { TrapAttackRenderer } from './renderer/trapAttackRenderer'
+
+export class TrapPlugin extends BaseAlchemyItemPlugin {
+  private attackRendererFactory?: TrapAttackRendererFactory
+  private trapPluginStore!: TrapPluginStore
+  private playerPluginStore!: PlayerPluginStore
+  private networkStore!: NetworkPluginStore<IMainScene>
+  private socketController?: SocketController
+  protected alchemyItem = TRAP_ITEM
+
+  public listenEvent(): void {
+    super.listenEvent()
+    this.bus.subscribeEvent('phaserSceneInit', this.phaserSceneInit.bind(this))
+    this.bus.subscribeEvent('phaserLoadAssets', this.loadAssets.bind(this))
+
+    this.bus.subscribeEvent('init', this.init.bind(this))
+
+    this.socketController = new SocketController(this.bus, this.store)
+    this.bus.subscribeEvent('registerMessage', this.socketController.registerMessage.bind(this.socketController))
+    this.bus.subscribeEvent(
+      'registerMessageListener',
+      this.socketController.setupRegisterMessageListener.bind(this.socketController)
+    )
+  }
+
+  private phaserSceneInit(ev: PhaserSceneInit): void {
+    this.attackRendererFactory = new TrapAttackRendererFactory(ev.scene)
+  }
+
+  private loadAssets(ev: PhaserLoadAssets): void {
+    TrapAttackRenderer.loadAssets(ev.scene)
+  }
+
+  private init(): void {
+    this.playerPluginStore = this.store.of('playerPlugin')
+    this.networkStore = this.store.of('networkPlugin')
+  }
+
+  protected subscribeGameEvent(): void {
+    super.subscribeGameEvent()
+    this.bus.subscribeEvent('useAlchemyItem', this.useAlchemyItem)
+    this.bus.subscribeEvent('entitySpawn', this.spawnTrap)
+    this.bus.subscribeEvent('entityDespawn', this.dieTrap)
+  }
+
+  protected unsubscribeGameEvent(): void {
+    super.unsubscribeGameEvent()
+    this.bus.unsubscribeEvent('useAlchemyItem', this.useAlchemyItem)
+    this.bus.unsubscribeEvent('entitySpawn', this.spawnTrap)
+    this.bus.unsubscribeEvent('entityDespawn', this.dieTrap)
+  }
+
+  protected handleGameStart(): void {
+    initTrapPluginStore(this.store, this.attackRendererFactory)
+    this.trapPluginStore = this.store.of('churarenTrapPlugin')
+    this.socketController?.getStores()
+    this.socketController?.registerMessageListener()
+  }
+
+  protected handleGameTermination(): void {
+    resetTrapPluginStore(this.store)
+    this.socketController?.unregisterMessageListener()
+    this.clearTrap()
```

Comment:
はじめにstoreを削除してしまうと、`this.clearTrap()`の際に、`undefined`が返ってくると思うので、修正したほうがよさそうです

```suggestion
    this.clearTrap()    this.socketController?.unregisterMessageListener()
    resetTrapPluginStore(this.store)
```